### PR TITLE
Term splits respect splitbelow, splitright

### PR DESCRIPTION
**Pull request by bnwa**
Perhaps a further enhancement would be to define the `direction` enum as `top | left | right | botton | float` and update logic accordingly, but this seemed to be the smallest change

### CODE REVIEW

**Comment by bnwa** [comment]
The expanded enum would permit the scenario wherein a user wants her buffer splits along one direction of an axis but the term to appear in the opposite direction but that seems niche

File: lua/toggleterm/ui.lua [hunk 1]
```diff
@@ -202,17 +202,29 @@ function M.switch_buf(buf)
...
-    new = "botright split",
-    resize = "resize",
-  },
-  vertical = {
-    existing = "rightbelow split",
-    new = "botright vsplit",
-    resize = "vertical resize",
-  },
+local function split_command(direction, existing)
+  if 'horizontal' == direction then
```
Comment (akinsho) [inline lua/toggleterm/ui.lua:206]: It's clear to me if this condition is reversed, i.e.
```suggestion
  if direction == 'horizontal' then
```

File: lua/toggleterm/ui.lua [hunk 2]
```diff
@@ -202,17 +202,29 @@ function M.switch_buf(buf)
...
-    existing = "rightbelow vsplit",
-    new = "botright split",
-    resize = "resize",
-  },
-  vertical = {
-    existing = "rightbelow split",
-    new = "botright vsplit",
-    resize = "vertical resize",
-  },
+local function split_command(direction, existing)
```
Comment (akinsho) [inline lua/toggleterm/ui.lua:205]: `existing` is a confusing name for this variable, I think `is_open` or `win_open` is a clearer name

File: lua/toggleterm/ui.lua [hunk 3]
```diff
@@ -202,17 +202,29 @@ function M.switch_buf(buf)
...
+  if 'horizontal' == direction then
+    if existing == true then
+      return"rightbelow vsplit"
+    elseif vim.o.splitbelow then
+      return "botright split"
+    else
+      return"topleft split"
```
Comment (akinsho) [inline lua/toggleterm/ui.lua:212]: there should be a space between the return and the string here

File: lua/toggleterm/ui.lua [hunk 4]
```diff
@@ -202,17 +202,29 @@ function M.switch_buf(buf)
...
-    resize = "resize",
-  },
-  vertical = {
-    existing = "rightbelow split",
-    new = "botright vsplit",
-    resize = "vertical resize",
-  },
+local function split_command(direction, existing)
+  if 'horizontal' == direction then
+    if existing == true then
```
Comment (akinsho) [inline lua/toggleterm/ui.lua:207]: There is no need to directly compare against `true` this should just be
```suggestion
    if existing then
```
The same goes for all the other `variable == true` unless there is a specific need to worry about comparing with some other truthy value vs true, then just use `if variable then`

File: lua/toggleterm/ui.lua [hunk 5]
```diff
@@ -273,17 +285,18 @@ end
...
     local split_win = win_ids[#win_ids]
+    local command = split_command(term.direction, true)
     if config.persist_size then M.save_window_size(term.direction, split_win) end
     api.nvim_set_current_win(split_win)
-    vim.cmd(commands.existing)
+    vim.cmd(command)
   else
-    vim.cmd(commands.new)
+    local command = split_command(term.direction, false)
+    vim.cmd(command)
```
Comment (akinsho) [inline lua/toggleterm/ui.lua:299]: There is no need for an interim variable here
```suggestion
    vim.cmd(split_command(term.direction, false))
```

**Review by akinsho — CHANGES_REQUESTED** [review]
@bnwa it would be good to have a clear description of what problem *exactly* you were facing that this PR solves. This plugin was initially designed to work exactly as it does to achieve the desired effect. I don't know how these changes will behave and whether they will be breaking for users who expect a specific kind of layout.
Would be good to get some clarification on exactly how these changes will impact how this plugin works

I've also made a few stylistic comments

**Comment by bnwa** [comment]
> @bnwa it would be good to have a clear description of what problem _exactly_ you were facing that this PR solves. 

As a user, I'd like to be able to control the direction of term splits. The code in your repo hardcoded the direction of splits in the `split_commands` table. For myself, it isn't ergonomic in my vim workflow to shift whatever I'm editing to the left for the sake of the terminal buffer, particularly as I leave them open during a session.

As I saw it, there were two choices then:
1. A more fine-grained revision to the `direction` enum, perhaps `top | left | bottom | right | float`, but this would force all users to update their config  and vim package managers have no especial means of notifying users of breaking changes
2. Update the handling of `direction` in the UI to respect `splitbelow` and `splitright` options; these options are already the users way of communicating their preference of which direction to open a new split along one of the two axes of `direction`.

However, I see now that I'd wrongly recalled that the neovim defaults for these options run contrary to your chosen split directions, which would be a major disruption for extant users.

So perhaps then a third option presents itself: a new config field indicating which (actual) direction along one of the two available axes represented in the current `direction` enum, defaulting to the extant behavior on `main`, would be better and less disruptive

**Pull request closed without merging**